tests/test_requirements.py

```diff
@@ -170,3 +170,8 @@ def test_sys_platform_linux_in(self):
         assert req.marker is not None
         assert req.marker.evaluate(dict(sys_platform="foo")) is True
         assert req.marker.evaluate(dict(sys_platform="bar")) is False
+
+    def test_parseexception_error_msg(self):
+        with pytest.raises(InvalidRequirement) as e:
+            Requirement("toto 42")
+            assert "Invalid Requiremnt" not in e
```

Comment:
Have you tried running this test without the changes you've made? I'm pretty sure it will still pass, the assertion will never be reached because the line before it raises the `InvalidRequirement` exception.

I also don't think you can assert whether a string is in an exception like that. It'd probably be better to assert that it's equivalent to an expected string, rather than asserting that it doesn't contain some string.

Reply:
Updated the test to check for a substring being in the exception now.